app/models/translation_cache.rb

```diff
@@ -36,12 +38,45 @@ def fetch_cache(cache_key, &block)
 
   def translations
     fetch_cache(cache_key) do
-      @model.send(@translation_attr_sym).to_a
-    end
+      @model.send(@translation_attr_sym).to_a.map { |model|
+        serialize(model)
+      }
```

Comment:
I cannot say I would fully understand this part of the code. Not a blocker for review.